controllers/kmsissuer_controller.go

```diff
@@ -84,44 +82,49 @@ func (r *KMSIssuerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 	// validation
 	if issuer.Spec.KeyID == "" {
-		return ctrl.Result{}, r.manageFailure(ctx, log, issuer, errors.New("INVALID KeyId"), fmt.Sprintf("Not a valid key: %s", issuer.Spec.KeyID))
+		return ctrl.Result{}, r.manageFailure(ctx, issuer, errors.New("INVALID KeyId"), fmt.Sprintf("Not a valid key: %s", issuer.Spec.KeyID))
 	}
 	// set default values
-	setIssuerDefaultValues(issuer)
+	r.setIssuerDefaultValues(issuer)
 
 	// Renew the certificate
 	certInput := desiredCertificateAuthorityCertificateInput(issuer)
 	desiredCert := r.KMSCA.GenerateCertificateAuthorityCertificate(certInput)
 
-	if certificateNeedsRenewal(issuer, desiredCert) {
+	if r.certificateNeedsRenewal(issuer, desiredCert) {
 		log.Info("generate certificate")
 		cert, err := r.KMSCA.GenerateAndSignCertificateAuthorityCertificate(certInput)
 		if err != nil {
-			return ctrl.Result{}, r.manageFailure(ctx, log, issuer, err, "Failed to generate the Certificate Authority Certificate")
+			return ctrl.Result{}, r.manageFailure(ctx, issuer, err, "Failed to generate the Certificate Authority Certificate")
 		}
 		issuer.Status.Certificate = pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw})
 		if err := r.Client.Status().Update(ctx, issuer); err != nil {
-			return ctrl.Result{}, r.manageFailure(ctx, log, issuer, err, "Failed to update the issuer with the issued Certificate")
+			return ctrl.Result{}, r.manageFailure(ctx, issuer, err, "Failed to update the issuer with the issued Certificate")
 		}
 	}
 	return ctrl.Result{
 		RequeueAfter: time.Until(desiredCert.NotAfter.Add(-1 * issuer.Spec.RenewBefore.Duration)),
-	}, r.manageSuccess(ctx, log, issuer)
+	}, r.manageSuccess(ctx, issuer)
 }
 
 // setIssuerDefaultValues
-func setIssuerDefaultValues(issuer *kmsiapi.KMSIssuer) {
+func (r *KMSIssuerReconciler) setIssuerDefaultValues(issuer *kmsiapi.KMSIssuer) {
+	log := r.Log.WithValues("name", issuer.Name, "namespace", issuer.Namespace)
 	if issuer.Spec.Duration == nil || issuer.Spec.Duration.Duration == 0 {
+		log.Info("setting default duration", "duration", DefaultCertDuration)
 		issuer.Spec.Duration = &metav1.Duration{Duration: DefaultCertDuration}
 	}
+	renewBefore := issuer.Spec.Duration.Duration * 2 / 3 //nolint:mnd // renew in 2/3 of the duration
```

Comment:
As mentioned in a previous comment, I think it's cleaner to have an internal constant than "magic number" in the code.

Reply:
fixed